archunit/src/test/java/com/tngtech/archunit/core/domain/JavaClassTest.java

```diff
@@ -439,7 +439,11 @@ public void direct_dependencies_from_self_by_member_declarations() {
                 .areAtLeast(2, parameterTypeDependency()
                         .from(AhavingMembersOfTypeB.class)
                         .to(B.class)
-                        .inLineNumber(0));
+                        .inLineNumber(0))
+                .areAtLeastOne(methodChecksInstanceOfDependency()
```

Comment:
I like it that you added it here :+1: :smiley: 
I wonder if we could test `>1` dependency in this place, since this assertion would fit in well here, and is missing otherwise?

Reply:
thx
What do you mean with "test `>1` dependency"?

Reply:
I mean more than one dependency. Because it seems like everywhere we're only checking if it works with one single dependency? In other words, if I would refactor it and only keep the last occurrence of an `instanceof` dependency I would expect all tests to pass, because we never check for more than one `instanceof` dependency anywhere. Or did I miss a place where this is already tested?

Reply:
ah, now I get it.
In contrast to the other dependency checks, the instanceof dependency contains a line number, which is also checked by the test. So to have a test that checks that there are at least 2 instanceof check dependencies, you would need to have the same dependency in the same line for the same classes. This seems weird and also because the dependency would be identical according to the `hashcode` and `equals` method, the dependency will only appear once in the `Set` returned from `getDirectDependenciesFromSelf`.
Another option would be to add a method `ignoringLineNumber` in `DependencyConditionCreation` to check dependencies independent of the line number.
I think due to checking the line number, the test is fine.

Reply:
Ah, yeah, you're right of course :joy: But you could have two checks of `.areAtLeastOne(methodChecksInstanceOfDependency()` including a different line number, right? That would also mean there is more than one `instanceof` dependency found...

Reply:
Done 😄